### PR TITLE
bwa-mem2 2.0

### PR DESCRIPTION
**Pull request by SeekingMeaning**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by kokyriakidis** [comment]
Can you add the new production version please?

**Comment by SeekingMeaning** [comment]
This patch fixes the `safe_mem_lib.h: No such file or directory` error:

```patch
-  url "https://github.com/bwa-mem2/bwa-mem2/archive/v2.0.tar.gz"
+  url "https://github.com/bwa-mem2/bwa-mem2.git",
+      :tag      => "v2.0",
+      :revision => "cbcc183c0843d20d45c84e066177eb8d58be2f9b"
```

However, this only results in another error:

```
ext/safestringlib/include/safe_mem_lib.h:100:16: error: conflicting types for 'memset_s'
extern errno_t memset_s(void *dest, rsize_t dmax, uint8_t value);
               ^
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/string.h:145:9: note: previous declaration is here
errno_t memset_s(void *__s, rsize_t __smax, int __c, rsize_t __n) __OSX_AVAILABLE_STARTING(__MAC_10_9, __IPHONE_7_0);
        ^
```

Because of this, bwa-mem2 2.0 is currently unable to build on macOS: intel/safestringlib#14

Bioconda "works around" this by not building for macOS at all: https://github.com/bioconda/bioconda-recipes/commit/f1f3925d6fc537a5493a0ca9b5c7afd4c4fab43e

File: Formula/bwa-mem2.rb [hunk 1]
```diff
@@ -12,6 +14,10 @@ class BwaMem2 < Formula
     sha256 "41f25256fd39920c9188bff511d422b926c9540c3aa8f6ed67c0e55aee25bc91" => :x86_64_linux
   end
 
+  # Only builds on Linux
+  # https://github.com/intel/safestringlib/issues/14
+  depends_on :linux
+
   uses_from_macos "zlib"
 
   def install
```
Comment (sjackman) [inline Formula/bwa-mem2.rb:23]:
```suggestion
  def install
    # Fix the error: error: conflicting types for 'memset_s'
    # See https://github.com/intel/safestringlib/issues/14
    inreplace "ext/safestringlib/include/safe_mem_lib.h", "extern errno_t memset_s", "//xxx extern errno_t memset_s"

```

File: Formula/bwa-mem2.rb [hunk 2]
```diff
@@ -12,6 +14,10 @@ class BwaMem2 < Formula
     sha256 "41f25256fd39920c9188bff511d422b926c9540c3aa8f6ed67c0e55aee25bc91" => :x86_64_linux
   end
 
+  # Only builds on Linux
+  # https://github.com/intel/safestringlib/issues/14
+  depends_on :linux
+
```
Comment (sjackman) [inline Formula/bwa-mem2.rb:20]: Please revert this chunk.

**Pull request merged**